tests/infrastructure/prometheus.go

```diff
@@ -508,6 +508,11 @@ var _ = DescribeSerialInfra("[rfe_id:3187][crit:medium][vendor:cnv-qe@redhat.com
 		}
 
 		for _, key := range keys {
+			// skip verification of metrics from previous migrations
+			if strings.Contains(key, "name=\"") && !stringContainsAny(key, nameMatchers) {
+				continue
+			}
```

Comment:
Aren't we just negating the `SatisfyAny` below?

Reply:
I thought about that but it makes sense. Maybe we should remove the `SatisfyAny` below, let me know what you think

The alternative would be changing the `collectMetrics` to be able to filter by a list of name maybe

Reply:
My main question is, since the test is serial, is it correct that the metrics of older migrations are there?

Reply:
Any migration that happened before can still show, it depends a lot on the time that passed since that migration.

```
23:10:27:   [FAILED] Expected
23:10:27:       <string>: kubevirt_vmi_migration_data_processed_bytes{name="testvmi-k7pqp",namespace="kubevirt-test-default1",node="node02"}
23:10:27:   To satisfy at least one of these matchers: [%!s(*matchers.ContainSubstringMatcher=&{name="%s" [testvmi-vscgz]}) %!s(*matchers.ContainSubstringMatcher=&{name="%s" [testvmi-6p5mb]})]
23:10:27:   In [It] at: tests/infrastructure/prometheus.go:518 @ 12/09/24 23:09:48.654
```

This is the error example I put in the PR description. And the metric has `name="testvmi-k7pqp"`  and the test expected `name="testvmi-vscgz"` or `name="testvmi-6p5mb"`. The test still validates that other metrics still contain these expected names. It really looks like a metric from other migrations

Reply:
Got it! I really think we should improve this test entirely.
Currently, it is too much dependant on external factors.
I think we should explicitly assert on which metrics we expect to be there and with which label and value the latter should come with.

Reply:
I think let's open a ticket to improve the test, but merge this pr to avoid the flakiness at the moment.
@fossedihelm WDYT?